### ISSUE TITLE
Enhancement: [prefer-nullish-coalescing] Allow ignorePrimitives option to be true

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#7180]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-nullish-coalescing

### Description

In #6487, @omril1 added a new `ignorePrimitives` option. Which is great! It lets us configure individually whether `bigint`s, `boolean`s, `number`s, and/or `string`s, are ignored by the rule.

As a convenience, it'd be nice to allow the option to be set as just the shorthand `true` - to set all of them as `true`.

```js
"@typescript-eslint/prefer-nullish-coalescing": ["error", {
  ignorePrimitives: true,
}]
```

### Fail

```typescript
n/a (no change)
```

### Pass

```typescript
n/a (no change)
```

### Additional Info

The label here is slightly inaccurate - it's not a new rule option, it's an augmentation to an existing option.

**Comment by omril1** [comment]
@JoshuaKGoldberg Doc link should to `prefer-nullish-coalescing`

**Comment by benneq** [comment]
Would this fix this issue i encountered after upgrading to 6.0.0 ?

```ts
const result = [].values().next();
// type of result = { done?: boolean | undefined }
    
if(result.done || Number.isFinite(42)) {
  // ...
}
```

It tells me to use `??` instead of `||`, but this would change the logic, because `done = false` wouldn't be handled the same as before:

```ts
false || true === true
false ?? true === false
```

Or is this considered a bug that should get its own ticket?

**Comment by bradzacher** [comment]
@benneq there is already an option for your case - please read the rule docs.
https://typescript-eslint.io/rules/prefer-nullish-coalescing/#ignoreprimitives
